cpp/tests/rolling/rolling_test.cpp

```diff
@@ -167,6 +168,253 @@ TEST_F(RollingStringTest, ZeroWindowSize)
+class RollingStructTest : public cudf::test::BaseFixture {
+};
+
+TEST_F(RollingStructTest, NoNullStructsMinMaxCount)
+{
+  using namespace cudf::test::iterators;
+  using strings_col = cudf::test::strings_column_wrapper;
+  using ints_col    = cudf::test::fixed_width_column_wrapper<int32_t>;
+  using structs_col = cudf::test::structs_column_wrapper;
+
+  auto const do_test = [](auto const& input) {
+    auto const expected_min = [] {
+      auto child1 = strings_col{
+        "This", "This", "being", "being", "being", "being", "column", "column", "column"};
+      auto child2 = ints_col{1, 1, 5, 5, 5, 5, 9, 9, 9};
+      return structs_col{{child1, child2}, no_nulls()};
+    }();
+
+    auto const expected_max = [] {
+      auto child1 = strings_col{
+        "rolling", "test", "test", "test", "test", "string", "string", "string", "string"};
+      auto child2 = ints_col{3, 4, 4, 4, 4, 8, 8, 8, 8};
+      return structs_col{{child1, child2}, no_nulls()};
+    }();
+
+    auto const expected_count = ints_col{{3, 4, 4, 4, 4, 4, 4, 3, 2}, no_nulls()};
+    auto constexpr preceeding = 2;
+    auto constexpr following  = 2;
+    auto constexpr min_period = 1;
+
+    auto const result_min =
+      cudf::rolling_window(input,
+                           preceeding,
+                           following,
+                           min_period,
+                           *cudf::make_min_aggregation<cudf::rolling_aggregation>());
+    auto const result_max =
+      cudf::rolling_window(input,
+                           preceeding,
+                           following,
+                           min_period,
+                           *cudf::make_max_aggregation<cudf::rolling_aggregation>());
+    auto const result_count_valid =
+      cudf::rolling_window(input,
+                           preceeding,
+                           following,
+                           min_period,
+                           *cudf::make_count_aggregation<cudf::rolling_aggregation>());
+    auto const result_count_all = cudf::rolling_window(
+      input,
+      preceeding,
+      following,
+      min_period,
+      *cudf::make_count_aggregation<cudf::rolling_aggregation>(cudf::null_policy::INCLUDE));
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_min, result_min->view());
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_max, result_max->view());
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_count, result_count_valid->view());
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_count, result_count_all->view());
+  };
+
+  auto const input_no_sliced = [] {
+    auto child1 =
+      strings_col{"This", "is", "rolling", "test", "being", "operated", "on", "string", "column"};
+    auto child2 = ints_col{1, 2, 3, 4, 5, 6, 7, 8, 9};
+    return structs_col{{child1, child2}};
+  }();
+
+  auto const input_before_sliced = [] {
+    auto constexpr dont_care{0};
+    // input_original needs to be static so it will not be destroyed.
```

Comment:
Maybe a deprecated comment? What is `input_original` refer to here?

Reply:
Yes, deprecated. Thanks for catching.